src/pthread-main.js

```diff
@@ -52,19 +52,39 @@ Module['print'] = threadPrint;
 Module['printErr'] = threadPrintErr;
 this.alert = threadAlert;
 
+Module['instantiateWasm'] = function(info, receiveInstance) {
+  // Instantiate from the module posted from the main thread.
+  // We can just use sync instantiation in the worker.
+  instance = new WebAssembly.Instance(Module['wasmModule'], info);
+  // We don't need the module anymore; new threads will be spawned from the main thread.
+  // XXX is this true?
```

Comment:
Yes, this is true, this comment can be removed. We don't want to chain Worker creations main thread -> Worker -> Worker, since we don't want to have deep chains, and main browser thread will then be able to message to all of them. So indeed in Worker the `Module['wasmModule']` bit is no longer needed. I'd prefer `delete Module['wasmModule']` over `Module['wasmModule'] = undefined;` however.